sim/src/caps.rs

```diff
@@ -39,6 +41,12 @@ impl Caps {
     pub fn get_num_images() -> usize {
         (unsafe { bootutil_get_num_images() }) as usize
     }
+
+    /// Query if this configuration performs some kind of upgrade by writing to flash.
+    pub fn modifies_flash() -> bool {
+        // All other configurations perform upgrades by writing to flash.
+        !(Self::RamLoad.present() || Self::DirectXip.present())
```

Comment:
In case of ram-load if image is faulty (could happen in negative test) the flash is erased:
https://github.com/mcu-tools/mcuboot/pull/974/commits/b005cde0550ec3fe25da6777e5335cae72a11918#diff-a004fdf5907914a360ce1e7da94221c0165d1c4c8ba7b8c4eaf67bd63af395c1R2863

Reply:
Maybe "modifies_flash" isn't the best name. I'm trying to capture the case of moving images around. Maybe 'moves_images' or something like that?

Reply:
@tamasban aside from that, I'm not certain that this behavior is actually desirable. If the external system does a partial download, and power were lost in the midst, it would have to start over. If power is unreliable, it might never complete. Or does it only do the erase if the image is marked for upgrade.

The only benefit I see to clearing a bad image would be to make subsequent boots faster.